examples/darwin-framework-tool/commands/delay/WaitForCommissioneeCommand.mm

```diff
@@ -69,6 +69,17 @@ - (void)device:(MTRDevice *)device receivedEventReport:(NSArray<NSDictionary<NSS
     auto * device = [MTRDevice deviceWithNodeID:@(mNodeId) controller:commissioner];
     VerifyOrReturnError(device != nil, CHIP_ERROR_INCORRECT_STATE);
 
+    auto state = [device state];
+    if (state == MTRDeviceStateReachable) {
```

Comment:
| The only way state can be something other than unknown is if we had the MTRDevice from earlier, right?
Yes.

| But then how would it be in any state other than MTRDeviceStateReachable here? Is the idea that if we did invalidateCASESession that can put us in unknown state synchronously? Because I don't think MTRDevice guarantees sync updates to reachability state or anything like that.

That was me not reading the code properly. My bad.

I have updated the code to not use 'MTRDevice'. 

| More generally, I don't think MTRDevice is the right tool for WaitForCommissioneeCommand, and the whole mExpireExistingSession is a crutch, right?

As far as I know we mostly did that to handle accessory 'Reboot' properly.

|  it might be simpler to just keep trying to issue reads for a must-exist attribute (e.g. FeatureMap of Descriptor on endpoint 0) to the device until we get a success response.

That is one possibility yes. I have to admit that this is nice to have a command that just try to set up a session though. For now I have used the old 'MTRDeviceCommissioner::getBaseDevice]` API in this patch, so the behaviour match `chip-tool` if needed someone may change it.